File: server-practice/message-server/hddc-application/src/main/kotlin/cool/hdd/application/command/service/CreatePostService.kt
package cool.hdd.application.command.service

import cool.hdd.application.command.dto.CreatePostDTO
import cool.hdd.application.command.ports.input.CreatePostUsecase
import cool.hdd.application.command.ports.out.CommandPostPort
import cool.hdd.domain.post.aggregate.PostAggregate
import cool.hdd.domain.post.service.PostCreationService
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional


@Service
@Transactional
class CreatePostService(
    private val commandPostPort: CommandPostPort,
) : CreatePostUsecase {

    override fun execute(dto: CreatePostDTO): PostAggregate =
        PostCreationService
            .execute(dto.convertPost()) {
                validateStatus()
            }
            .saved()

}

File: server-practice/py-server/py_application/post/service/create_post_service.py
from py_application.post.ports.input.create_post_usecase import CreatePostUsecase, CreatePostDTO
from py_application.post.ports.output.post_command_port import PostCommandPort
from py_domain.post.aggregate.post_aggregate import PostAggregate
from py_domain.post.entity.post import Post

class CreatePostService(CreatePostUsecase):
    def __init__(self, post_command_port: PostCommandPort):
        self._post_command_port = post_command_port

    def create(self, dto: CreatePostDTO) -> Post:
        aggregate = PostAggregate.create(
            title=dto.title,
            content=dto.content
        )
        saved_aggregate = self._post_command_port.save(aggregate)
        return saved_aggregate.post

